app/src/store/playground/playgroundStore.tsx

```diff
@@ -290,11 +288,55 @@ export const createPlaygroundStore = (
           }
         }
       });
-      set({ input: { variables: { ...variables } } });
+      set({
+        input: { ...get().input, variableKeys: [...Array.from(variables)] },
+      });
+    },
+    setVariableValue: (key: string, value: string) => {
+      const input = get().input;
+      if ("variablesValueCache" in input) {
+        set({
+          input: {
+            ...input,
+            variablesValueCache: { ...input.variablesValueCache, [key]: value },
+          },
+        });
+      }
     },
```

Comment:
why would this not be present? it's the only place where variable values can be set correct?

Reply:
It's because we have `input` typed as a union 

```
type DatasetInput = {
  datasetId: string;
};

type ManualInput = {
  variablesValueCache: Record<string, string>;
  variableKeys: string[];
};

type PlaygroundInput = DatasetInput | ManualInput;
```

Meaning I have to narrow what could be inside `input` by testing its keys

Reply:
oh are these basically typeguards against dataset input type? if so maybe would just make a typeguard and use it in all these places

Reply:
Refactored into type guards